web/app.ts

```diff
@@ -66,6 +67,25 @@ function startRecording(microphone: MediaStream) {
 
 async function onRecordDataAvailable(blob: Blob) {
   const base64Data = (await blobToBase64(blob)) as string;
+  dataToSend.push(base64Data);
+}
+
+console.log("init");
+
+initE2E("demo", "demo").then((id) => {
+  console.log(`got session id ${id}`);
+  SESSION_ID = id;
+  captureMicrophone(function (mic) {
+    startRecording(mic);
+  });
+});
+
+async function sendData() {
+  const base64Data = dataToSend.shift();
```

Comment:
Shall we add comments about why we are doing this?
```suggestion
  //Speech data has to be sent in order.
  const base64Data = dataToSend.shift();
```